adc/ade7913/ade7913-driver.c

```diff
@@ -605,131 +832,145 @@ static void msg_loop(void)
 			break;
 		}
+	} while (0);
 
-		if ((res = dev_init()) < 0) {
-			log_error("device initialization failed");
-			resourceDestroy(common.gpio_irq_lock);
-			resourceDestroy(common.gpio_irq_cond);
-			resourceDestroy(common.edma_spi_rcv_lock);
+	if (res < 0) {
+		if (common.oid.port != (uint32_t)-1) {
+			portDestroy(common.oid.port);
+		}
+		if (common.edma_spi_rcv_lock != (handle_t)-1) {
 			resourceDestroy(common.edma_spi_rcv_lock);
+		}
+		if (common.edma_spi_rcv_cond != (handle_t)-1) {
+			resourceDestroy(common.edma_spi_rcv_cond);
+		}
+		if (common.dready_cond != (handle_t)-1) {
 			resourceDestroy(common.dready_cond);
+		}
+		if (common.buff != MAP_FAILED) {
 			munmap(common.buff, (ADC_BUFFER_SIZE + _PAGE_SIZE - 1) / _PAGE_SIZE * _PAGE_SIZE);
-
-			break;
 		}
-
-		/* Enable SPI edma receive request */
-		*(common.spi1_ptr + spi_der) = (1 << 1);
-		/* Enable GPIO DREADY interrupt */
-		*(common.gpio3_ptr + gpio_imr) |= (0b1 << 20);
-	} while (0);
+	}
 
 	return res;
 }
 
 
 int main(int argc, char **argv)
 {
-	int i;
-	oid_t root;
+	oid_t tmp_oid;
+	int i, err;
+
+	priority(ADE7913_PRIO);
 
-	common.gpio3_ptr = (void *)0x40134000;
-	common.spi1_ptr = (void *)0x40114000;
+	memset(&common, 0, sizeof(common));
 
 	if (argc != 3) {
 		log_error("No device list or no SPI device given");
-		return -EINVAL;
+		return EXIT_FAILURE;
 	}
-	else {
-		common.order = argv[1];
-		common.devcnt = strlen(argv[1]);
-		common.spi = atoi(argv[2]);
-
-		for (i = 0; i < common.devcnt; ++i) {
-			if ((int)(common.order[i] - '0') >= common.devcnt || (int)(common.order[i] - '0') < 0) {
-				log_error("Wrong order format provided");
-				return -1;
-			}
-		}
 
-		if (common.spi < 0) {
-			log_error("Wrong spi number provided");
-			return -1;
-		}
+	common.spi = atoi(argv[2]);
+	if (common.spi < 1 || common.spi > sizeof(spi_base) / sizeof(spi_base[0])) {
+		log_error("Wrong spi number provided");
+		return EXIT_FAILURE;
+	}
 
-		log_info("Device order: %s", common.order);
+	common.oid.port = (uint32_t)-1;
+	common.edma_spi_rcv_lock = (handle_t)-1;
+	common.edma_spi_rcv_cond = (handle_t)-1;
+	common.dready_cond = (handle_t)-1;
+	common.buff = MAP_FAILED;
+
+	common.spi_ptr = (uint32_t *)spi_base[common.spi - 1];
+	common.gpio3_ptr = (uint32_t *)0x40134000;
+	common.iomux_ptr[0] = (uint32_t *)0x400e8000 + 4 + pctl_mux_gpio_ad_18;
+	common.iomux_ptr[1] = (uint32_t *)0x400e8000 + 4 + pctl_mux_gpio_ad_19;
+	common.iomux_ptr[2] = (uint32_t *)0x400e8000 + 4 + pctl_mux_gpio_ad_20;
+	common.iomux_ptr[3] = (uint32_t *)0x400e8000 + 4 + pctl_mux_gpio_ad_29;
+
+	common.order = argv[1];
+	common.devcnt = strlen(common.order);
+
+	if (common.devcnt < 1 || common.devcnt > 4) {
+		log_error("Incorrect ADE7913 device count (1 min, 4 max)");
+		return EXIT_FAILURE;
 	}
 
-	if (common.devcnt > 4) {
-		log_error("Incorrect ADE7913 device count (4 max)");
-		return -EINVAL;
+	for (i = 0; i < common.devcnt; ++i) {
+		int devnum = common.order[i] - '0';
```

Comment:
I know it's a good thing, but this code in general does not use reduced scope variables and it's still not acknowledged by our coding convention

Reply:
Ok, You're right.